File: frontend/src/types/session.ts
export interface SessionSummary {
  id: string;
  status: string;
  current_stage: string | null;
  progress: number;
  created_at: string;
  expires_at: string | null;
  last_activity_at: string;
}

export interface SessionDetail extends SessionSummary {
  config: Record<string, unknown>;
  documents: import('./document').Document[];
}

export interface SessionListResponse {
  items: SessionSummary[];
  pagination: {
    page: number;
    page_size: number;
    total: number;
  };
}

export interface SessionCreateRequest {
  document_ids: string[];
  config?: Record<string, unknown>;
  created_by?: string | null;
}

export interface SessionCreateResponse {
  session_id: string;
  status: string;
  expires_at: string | null;
}

export interface SessionResultsResponse {
  analysis: Record<string, unknown> | string;
  test_cases: Record<string, unknown>;
  statistics: Record<string, unknown>;
  version: number;
  generated_at: string;
}

File: backend/app/schemas/session.py
"""Pydantic schemas for session management."""

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field

from app.schemas.document import DocumentOut


class PaginationMeta(BaseModel):
    page: int = 1
    page_size: int = 20
    total: int = 0


class SessionCreateRequest(BaseModel):
    document_ids: list[str] = Field(..., min_length=1)
    config: dict[str, Any] = Field(default_factory=dict)
    created_by: str | None = None


class SessionConfirmationRequest(BaseModel):
    stage: str
    decision: str
    comment: str | None = None


class SessionSummary(BaseModel):
    id: str
    status: str
    current_stage: str | None
    progress: float
    created_at: datetime
    expires_at: datetime | None
    last_activity_at: datetime

    model_config = {"from_attributes": True}


class SessionDetail(SessionSummary):
    config: dict[str, Any]
    documents: list[DocumentOut]

    model_config = {"from_attributes": True}


class SessionListResponse(BaseModel):
    items: list[SessionSummary]
    pagination: PaginationMeta


class SessionCreateResponse(BaseModel):
    session_id: str
    status: str
    expires_at: datetime | None


class SessionResultsResponse(BaseModel):
    analysis: dict[str, Any] | str
    test_cases: dict[str, Any]
    statistics: dict[str, Any]
    version: int
    generated_at: datetime


class ExportRequest(BaseModel):
    result_version: int | None = None
